transformer/switch.go

```diff
@@ -0,0 +1,70 @@
+/*
+ * skogul, switch transformer
+ *
+ * Copyright (c) 2019 Telenor Norge AS
+ * Author(s):
+ *  - Håkon Solbjørg <hakon.solbjorg@telenor.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301  USA
+ */
+
+package transformer
+
+import (
+	"github.com/telenornms/skogul"
+)
+
+// Case requires a field ("when") and a value ("is") to match
+// for the set of transformers to run
+type Case struct {
+	When         string                  `doc:"Used as a conditional statement on a field"`
+	Is           string                  `doc:"Used for the specific value (string) of the stated metadata field"`
+	Transformers []skogul.TransformerRef `doc:"The transformers to run when the defined conditional is true"`
+}
+
```

Comment:
This is rather interesting. I don't think we expose nested doc strings, but we probably should. At the moment, this probably means that the end-user documentation is a bit lacking, but I'm inclined to accept that, and instead raise a separate issue for config/help.go to detect this and expose nested documentation/custom types. Shouldn't be TOO hard, and I'd rather do that than limit our options in how we code modules.

Reply:
More docs == better? :D It gives me documentation in my editor, at least, so that's nice.